internal/usecase/crypto.go

```diff
@@ -156,6 +201,255 @@ func (uc *CryptoUsecase) GetDecryptionKey(ctx context.Context, identity string)
+		if err != nil {
+			log.Err(err).Msg("err encountered while generating identity prefix")
+			err := httpError.NewHttpError(
+				"error encountered while generating identity prefix",
+				"",
+				http.StatusInternalServerError,
+			)
+			return nil, &err
+		}
+		identityPrefix = block
+	}
+
+	blockNumber, err := uc.ethClient.BlockNumber(ctx)
+	if err != nil {
+		log.Err(err).Msg("err encountered while querying for recent block")
+		err := httpError.NewHttpError(
+			"error encountered while querying for recent block",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	eon, err := uc.keyperSetManagerContract.GetKeyperSetIndexByBlock(nil, blockNumber)
+	if err != nil {
+		log.Err(err).Msg("err encountered while querying keyper set index")
+		err := httpError.NewHttpError(
+			"error encountered while querying for keyper set index",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	eonKeyBytes, err := uc.keyBroadcastContract.GetEonKey(nil, eon)
+	if err != nil {
+		log.Err(err).Msg("err encountered while querying for eon key")
+		err := httpError.NewHttpError(
+			"error encountered while querying for eon key",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	eonKey := &shcrypto.EonPublicKey{}
+	if err := eonKey.Unmarshal(eonKeyBytes); err != nil {
+		log.Err(err).Msg("err encountered while deserializing eon key")
+		err := httpError.NewHttpError(
+			"error encountered while querying deserializing eon key",
+			"",
+			http.StatusInternalServerError,
+		)
+		return nil, &err
+	}
+
+	identity := common.ComputeIdentity(identityPrefix[:], ethCommon.HexToAddress(address))
+
+	return &GetDataForEncryptionResponse{
+		Eon:            eon,
+		Identity:       hex.EncodeToString(identity.Marshal()),
+		IdentityPrefix: hex.EncodeToString(identityPrefix[:]),
+		EonKey:         hex.EncodeToString(eonKeyBytes),
+	}, nil
+}
+
+func (uc *CryptoUsecase) RegisterIdentity(ctx context.Context, decryptionTimestamp uint64, identityPrefixStringified string) (*RegisterIdentityResponse, *httpError.Http) {
+	currentTimestamp := time.Now().Unix()
+	if currentTimestamp > int64(decryptionTimestamp) {
+		log.Debug().Uint64("decryptionTimestamp", decryptionTimestamp).Int64("currentTimestamp", currentTimestamp).Msg("decryption timestamp should be in future")
+		err := httpError.NewHttpError(
+			"decryption timestamp should be in future",
+			"",
+			http.StatusBadRequest,
+		)
+		return nil, &err
+	}
+	var identityPrefix shcrypto.Block
+
+	if len(identityPrefixStringified) > 0 {
+		identityPrefixBytes, err := hex.DecodeString(strings.TrimPrefix(identityPrefixStringified, "0x"))
+		if err != nil {
+			log.Err(err).Msg("err encountered while decoding identity prefix")
+			err := httpError.NewHttpError(
+				"error encountered while decoding identity prefix",
+				"",
+				http.StatusBadRequest,
+			)
+			return nil, &err
+		}
+		if len(identityPrefixBytes) != 32 {
```

Comment:
Same comment here, would be better to move the length check before the decoding to avoid unnecessary decoding